vllm/prefix.py

```diff
@@ -44,43 +47,111 @@ def get_length(self) -> int:
     def __hash__(self) -> int:
         return self.hash
 
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, Prefix):
+            return False
+        return self.hash == other.hash
+
     def set_block_table(self, block_table: BlockTable) -> None:
         self.block_table = block_table.copy()
 
 
 class PrefixPool:
-    """Manages all the prompt prefixes.
-
-    NOTE: This feature is experimental and may be replaced with automatic
-        prefix caching in the future.
+    """Manages all the prompt prefixes. If the max_capacity argument is not None,
+    the pool will act as a LRU cache and remove the least recently used prefix once
+    the capacity is reached.
 
     Args:
         block_size: The block size of the executed model.
 
     Attributes:
         prefixes: A list of all the prefixes.
         block_size: The block size of the executed model.
+        max_capacity: The maximum number of prefixes allowed in the pool at any given time. 
+            The default value is None, which means there is no limit. It can only take positive
+            values.
     """
 
-    def __init__(
-        self,
-        block_size: int,
-    ) -> None:
-        # TODO(zhuohan): Add a capacity limit to the prefix pool.
-        self.prefixes: Dict[int, Prefix] = {}
+    def __init__(self,
+                 block_size: int,
+                 max_capacity: Optional[int] = None) -> None:
+        self.prefixes: OrderedDict[int, Prefix] = OrderedDict()
         self.block_size = block_size
 
+        if max_capacity is not None:
+            # NOTE(to remove after consultation): I have also been thinking if we need to
```

Comment:
The thoughts of these comments are no longer relevant given that the `.get_prefixes_to_free()` function has a mechanism to check that no weird condition arises. 
TODO: Remove these comments.

Reply:
Removed.

Reply:
[c04fe27](https://github.com/vllm-project/vllm/pull/2511/commits/c04fe27abd3a10e55560e6612b8c8f09943ff8cf)